### PR TITLE
Bug 1133146 support multiple firefox os versions

### PR DESCRIPTION
**Pull request by alexgibson**
Still waiting on some additional countries being added, so **do not merge** for now. But this is good for code review!

### CODE REVIEW

**Comment by pmclanahan** [comment]
@jgmize like Alex said, we're just waiting for some country clarification. Please r? the python bits when you get time.

File: bedrock/firefox/views.py [hunk 1]
```diff
@@ -250,17 +249,15 @@ def all_downloads(request, channel):
...
-    lang_file = 'firefox/os/index-new'
-    old_home = 'firefox/os/index.html'
-    new_home = 'firefox/os/index-new.html'
+@never_cache
+def firefox_os_geo_redirect(request):
+    country = get_country_from_request(request)
+    try:
+        version = settings.FIREFOX_OS_COUNTRY_VERSIONS[country]
+    except KeyError:
+        version = settings.FIREFOX_OS_COUNTRY_VERSIONS['default']
```
Comment (jgmize) [inline bedrock/firefox/views.py]: NIT: I personally prefer the `get` method with a default over exception handling. How about something like this instead?

```python
version = settings.FIREFOX_OS_COUNTRY_VERSIONS.get(
    country,
    settings.FIREFOX_OS_COUNTRY_VERSIONS['default'])
```

**Comment by jgmize** [comment]
Other than one tiny nit which you can feel free to ignore, the python bits lgtm. :+1:

File: bedrock/mozorg/templates/mozorg/robots.txt [hunk 2]
```diff
@@ -6,5 +6,9 @@ disallow: /*/download/
 disallow: /*/firstrun/
 disallow: /*/newsletter/existing/
 disallow: /*/whatsnew/
+disallow: /*/firefox/os/1.1/
```
Comment (alexgibson) [inline bedrock/mozorg/templates/mozorg/robots.txt]: @pmclanahan - added these to `robots.txt` as suggested. Just checking these look ok?

Comment (pmclanahan) [inline bedrock/mozorg/templates/mozorg/robots.txt]: I saw that. Looks good!

File: bedrock/firefox/views.py (hunk 1 above)
Comment (pmclanahan) [inline bedrock/firefox/views.py]: Yeah. I waffled on that myself. I'll try it the other way.

**Comment by alexgibson** [comment]
@craigcook r?

**Comment by alexgibson** [comment]
Also ping @flodolo - there are two new strings here to update the page title & description (again, sorry). 

And also [this string](https://github.com/mozilla/bedrock/pull/2866/files#diff-1df3cc28c78c6d328f3b94934753903fR19) is no longer hidden behind an l10n tag. Hopefully it's translated by now, not quite sure what else we could revert to in this case.

**Comment by flodolo** [comment]
I will add the 2 new strings, it's ok to leave the other one not behind tag, it affects only 4 locales out of 21, and they should catch up.

**Comment by alexgibson** [comment]
@pmclanahan - looks like some of the views/tests need a rebase. I can take a stab at it, but perhaps you might be best in case I miss something that got re-ordered?

**Comment by pmclanahan** [comment]
@alexgibson I'll do the rebase now and force push the branch. You're not making further changes right?

**Comment by alexgibson** [comment]
@pmclanahan - go for it, thanks! :+1:

**Comment by pmclanahan** [comment]
Rebase done. Should be good to go! :8ball:

**Comment by craigcook** [comment]
Looks good on the front-end. r+ :hamburger: 

Should we squash commits, or are they distinct enough to keep separate?

**Comment by pmclanahan** [comment]
Commits are fine w/ me as they are.

**Pull request merged**